src/circuit/circuit_frame.rs

```diff
@@ -4656,36 +4653,62 @@ pub fn enforce_at_most_n_bits<F: LurkField, CS: ConstraintSystem<F>>(
     Ok(())
 }
 
-#[derive(Copy, Clone)]
+// Enforce v is the bit decomposition of num, therefore we have that 0 <= num < 2ˆ(sizeof(v)).
+pub fn enforce_pack<F: LurkField, CS: ConstraintSystem<F>>(
+    mut cs: CS,
+    v: &[Boolean],
+    num: AllocatedNum<F>,
+) -> Result<(), SynthesisError> {
+    let mut coeff = F::one();
+
+    let mut v_lc = LinearCombination::<F>::zero();
+    for i in 0..v.len() {
+        match v[i] {
+            Boolean::Constant(c) => {
+                if c {
+                    v_lc = v_lc + (coeff, CS::one())
+                }
+            }
+            Boolean::Is(ref v) => v_lc = v_lc + (coeff, v.get_variable()),
+            Boolean::Not(ref v) => v_lc = v_lc + (coeff, CS::one()) - (coeff, v.get_variable()),
+        };
+        coeff = coeff.double();
+    }
+
+    cs.enforce(
+        || "pack",
+        |_| v_lc,
+        |lc| lc + CS::one(),
+        |lc| lc + num.get_variable(),
+    );
+
+    Ok(())
+}
+
+// Lurk supported uint types
 pub enum UnsignedType {
     U32,
     U64,
 }
 
-// Convert from num to u32.
-pub fn to_unsigned_int<F: LurkField, CS: ConstraintSystem<F>>(
+pub fn to_unsigned_integer_helper<F: LurkField, CS: ConstraintSystem<F>>(
     mut cs: CS,
     g: &GlobalAllocations<F>,
-    maybe_unsigned: &AllocatedNum<F>,
+    field_elem: &AllocatedNum<F>,
+    field_bn: BigUint,
+    field_bits: Vec<Boolean>,
```

Comment:
This should probably be of type `&[Boolean]` (which would let you avoid cloning the vector below).

Reply:
done